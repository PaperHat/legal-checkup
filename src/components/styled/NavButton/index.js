import styled from 'styled-components'

const NavButton = styled.button`
  background-color: #ededed;
  border: none;
  color: ${({ onClick }) => (onClick ? '#000' : '#c0c0c0')};
  height: 42px;
  outline: none;
  width: 48px;
`

NavButton.displayName = 'StyledNavButton'

export default NavButton
