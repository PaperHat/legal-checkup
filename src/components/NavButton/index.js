import React from 'react'

import { StyledNavButton } from '../styled'

export default function NavButton ({ answered, current, type }) {
  if (type === 'back') {
    return current > 1 ? (
      <StyledNavButton type={type}>&lt;</StyledNavButton>
    ) : (
      <StyledNavButton type={type} disabled>
        &lt;
      </StyledNavButton>
    )
  }

  if (type === 'forward') {
    return current < answered ? (
      <StyledNavButton type={type}>&gt;</StyledNavButton>
    ) : (
      <StyledNavButton type={type} disabled>
        &gt;
      </StyledNavButton>
    )
  }

  return <StyledNavButton />
}
