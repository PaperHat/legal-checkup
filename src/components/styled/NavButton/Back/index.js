import NavButton from '..'

const NavButtonBack = NavButton.extend`
  border-radius: 5px 0 0 5px;
`

NavButtonBack.displayName = 'StyledNavButtonBack'

export default NavButtonBack
