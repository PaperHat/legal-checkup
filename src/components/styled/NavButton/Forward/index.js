import NavButton from '..'

const NavButtonForward = NavButton.extend`
  border-radius: 0 5px 5px 0;
`
NavButtonForward.displayName = 'StyledNavButtonForward'

export default NavButtonForward
