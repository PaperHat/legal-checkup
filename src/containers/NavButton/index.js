import { connect } from 'react-redux'

import NavButton from '../../components/NavButton'
import {
  getActiveQuestion,
  getCount,
  nextQuestion,
  previousQuestion
} from '../../state'

function mapStateToProps (state, { direction }) {
  const activeQuestion = getActiveQuestion(state)
  const answeredQuestions = getCount(state)
  const unlocked =
    direction === 'back'
      ? activeQuestion > 1
      : direction === 'forward' ? activeQuestion < answeredQuestions + 1 : false

  return {
    unlocked
  }
}

function mapDispatchToProps (dispatch, { direction }) {
  const action =
    direction === 'back'
      ? previousQuestion()
      : direction === 'forward' ? nextQuestion() : {}

  return {
    handleClick: () => dispatch(action)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(NavButton)
