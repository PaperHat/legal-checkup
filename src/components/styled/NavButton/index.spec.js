import React from 'react'
import { shallow } from 'enzyme'

import NavButton from '.'

describe('components:styled:NavButton', () => {
  it('renders the NavButton and CSS properties properly', () => {
    expect(toJson(shallow(<NavButton>&#x25C0;</NavButton>))).toMatchSnapshot()
  })

  it('renders the NavButton and CSS properties properly with onClick prop', () => {
    expect(
      toJson(shallow(<NavButton onClick={() => null}>&#x25C0;</NavButton>))
    ).toMatchSnapshot()
  })
})
