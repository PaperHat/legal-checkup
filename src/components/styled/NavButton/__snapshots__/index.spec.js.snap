// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:styled:NavButton renders the NavButton and CSS properties properly 1`] = `
.c0 {
  background-color: #ededed;
  border: none;
  color: #c0c0c0;
  height: 42px;
  outline: none;
  width: 48px;
}

<button
  className="c0"
>
  ◀
</button>
`;

exports[`components:styled:NavButton renders the NavButton and CSS properties properly with onClick prop 1`] = `
.c0 {
  background-color: #ededed;
  border: none;
  color: #000;
  height: 42px;
  outline: none;
  width: 48px;
}

<button
  className="c0"
  onClick={[Function]}
>
  ◀
</button>
`;
